### DISCUSSION TITLE
@adonisjs/redis - Support multiple subscriptions to one Redis channel

### DISCUSSION THREAD

**Discussion by jonahsnider** [adonisjs/core#4640] [Ideas]
In @adonisjs/redis, trying to do `redis.subscribe()` when the specified channel already has an active subscription [will throw an error](https://github.com/adonisjs/redis/blob/8432ff020e28a5735c5c6f6689a8fbea2408f531/src/connections/abstract_connection.ts#L285).

It would be nice if instead of throwing, @adonisjs/redis was able to intelligently group duplicate subscriptions together.

```js
redis.subscribe('users/123', callback); // opens connection

// later...
redis.subscribe('users/123', callback); // reuses existing subscription
```

@adonisjs/redis is described as "The package is a thin wrapper on top of ioredis with better DX around Pub/Sub and automatic management of multiple redis connections", so this feature seems very inline with the package's goals.

**Comment by thetutlage** [comment]
@jonahsnider Yup, I purposely did not add support for it, waiting to see if it is a common use-case and worth supporting or not.

Can you please open an issue with the `adonisjs/redis` repo and we can work on it. Also, feel free to open a PR, if you think you can write the implementation.

**Reply by jonahsnider** [reply]
Sure! I originally went to open an issue there but the only allowed template is for a bug report:
![Screenshot of GH issue labels](https://github.com/adonisjs/core/assets/7608555/24ef4960-e1aa-4846-a36d-4f7844846cfb)


Should I just fill that out and have you/someone re-label it after it's created?

**Reply by thetutlage** [reply]
There is one for "Feature Request"

**Reply by thetutlage** [reply]
Ohh okay. Ignore the last comment. Yes, go with "Bug report" for now. But I think, we should allow a direct issue for Feature requests too.

**Comment by jonahsnider** [comment]
Discussion moved to https://github.com/adonisjs/redis/issues/65
